android/src/main/kotlin/io/didomi/didomi_sdk/DidomiSdkPlugin.kt

```diff
@@ -0,0 +1,128 @@
+package io.didomi.didomi_sdk
+
+import android.app.Activity
+import android.content.Intent
+import androidx.annotation.NonNull
+import io.didomi.sdk.Didomi
+import io.didomi.sdk.Log
+import io.flutter.embedding.engine.plugins.FlutterPlugin
+import io.flutter.plugin.common.MethodCall
+import io.flutter.plugin.common.MethodChannel
+import io.flutter.plugin.common.MethodChannel.MethodCallHandler
+import io.flutter.plugin.common.MethodChannel.Result
+import io.flutter.plugin.common.PluginRegistry.Registrar
+import io.flutter.embedding.engine.plugins.activity.ActivityAware
+import io.flutter.embedding.engine.plugins.activity.ActivityPluginBinding
+import androidx.fragment.app.FragmentActivity
+
+/** DidomiSdkPlugin */
+class DidomiSdkPlugin : FlutterPlugin, MethodCallHandler, ActivityAware {
+    /// The MethodChannel that will the communication between Flutter and native Android
+    ///
+    /// This local reference serves to register the plugin with the Flutter Engine and unregister it
+    /// when the Flutter Engine is detached from the Activity
+    private lateinit var channel: MethodChannel
+
+    /// The Activity used to interact with the Didomi SDK.
+    ///
+    /// This is always the current displayed Activity, as activities can not be passed through Flutter channels
+    private var currentActivity: Activity? = null
+
+    override fun onAttachedToEngine(@NonNull flutterPluginBinding: FlutterPlugin.FlutterPluginBinding) {
+        channel = MethodChannel(flutterPluginBinding.binaryMessenger, "didomi_sdk")
+        Log.level = android.util.Log.VERBOSE
+
+        channel.setMethodCallHandler(this)
+    }
+
+    override fun onAttachedToActivity(activityPluginBinding: ActivityPluginBinding) {
+        // Update the current activity
+        activityPluginBinding.activity?.also {
+            currentActivity = it
+        }
+    }
+
+    override fun onDetachedFromEngine(@NonNull binding: FlutterPlugin.FlutterPluginBinding) {
+        channel.setMethodCallHandler(null)
+    }
+
+    override fun onDetachedFromActivity() {
+        // Release the current activity
+        currentActivity = null
+    }
+
+    override fun onDetachedFromActivityForConfigChanges() {
+        onDetachedFromActivity()
+    }
+
+    override fun onReattachedToActivityForConfigChanges(binding: ActivityPluginBinding) {
+        onAttachedToActivity(binding)
+    }
+
+    override fun onMethodCall(@NonNull call: MethodCall, @NonNull result: Result) {
+        try {
+            when (call.method) {
+
+                "getPlatformVersion" -> result.success("Android ${android.os.Build.VERSION.RELEASE}")
+
+                "initialize" -> initializeSdk(call, result)
+
+                "setupUI" -> {
+                    getFragmentActivity(result)?.also {
+                        Didomi.getInstance().setupUI(it)
+                        result.success(null)
+                    }
```

Comment:
Maybe we should throw a specific message in case fragmentManager is unavailable?